src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -504,68 +506,63 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
-                    logger.atDebug().log("The fileHash does not exist in directory");
-                    return;
-                }
-                file = file1.get();
-                fileName = file.getAbsolutePath();
-                // TODO: the following logic is only for passing this small PR while not changing the current context.
-                //  We will grab the fileHash in the future.
-                Optional<LogFile> activeFile = logFileGroup.getActiveFile();
-                if (activeFile.isPresent()) {
-                    isActiveFile = activeFile.get().fileEquals(file);
-                }
-            } catch (InvalidLogGroupException e) {
-                logger.atDebug().cause(e).log();
+        try {
+            LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
+            if (!logFileGroup.getFile(fileHash).isPresent()) {
+                logger.atTrace().log("The fileHash does not exist in directory");
                 return;
             }
-        }
-        // If we have completely read the file, then we need add it to the completed files list and remove it
-        // it (if necessary) for the current processing list.
-        String componentName = attemptLogInformation.getComponentName();
-        if (!isActiveFile && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
-                + cloudWatchAttemptLogFileInformation.getStartPosition()) {
-            Set<String> completedFileNames = completedLogFilePerComponent.getOrDefault(componentName, new HashSet<>());
-            completedFileNames.add(fileName);
-            completedLogFilePerComponent.put(componentName, completedFileNames);
-            if (currentProcessingLogFilePerComponent.containsKey(componentName)) {
-                CurrentProcessingFileInformation fileInformation = currentProcessingLogFilePerComponent
-                        .get(componentName);
-                if (fileInformation.fileName.equals(fileName)) {
-                    currentProcessingLogFilePerComponent.remove(componentName);
+            LogFile file = logFileGroup.getFile(fileHash).get();
+            Optional<LogFile> activeFile = logFileGroup.getActiveFile();
+            boolean isActiveFile = false;
+            if (activeFile.isPresent()) {
+                isActiveFile = activeFile.get().fileEquals(file);
+            }
+            //This is deprecated value in versions greater than 2.2, but keep it here to avoid
+            // upgrade-downgrade issues.
+            String fileName = activeFile.get().getAbsolutePath();
+            // If we have completely read the file, then we need add it to the completed files list and remove it
+            // it (if necessary) for the current processing list.
+            String componentName = attemptLogInformation.getComponentName();
+            if (!isActiveFile && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
```

Comment:
Add a method to the LogFile returned by the log group that you can use to check if the file is the active one or not

```suggestion
            if (!logFile.isActive() && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
```